### PR TITLE
[CRT/ARM] Fix Implementations of __rt_udiv and __rt_sdiv

### PR DESCRIPTION
**Pull request by tkreuzer**
Also update file headers / switch to MIT license

### CODE REVIEW

**Review by binarymaster — COMMENTED** [review]
Thanks for the fix! I tested it, does not crash anymore, however udiv64/sdiv64 functions don't work properly, see my test code and debug results.

I used this code to test division functions, by integrating them into rapps:
```c
ULONGLONG udividend64 = 0x2AFFFFFFF * 100;
LONGLONG sdividend64 = -0x2AFFFFFFF * 100;
ULONG udividend = 3425;
LONG sdividend = -3425;
ULONG divisor = 400;
LONG sdivisor = 400;
ULONGLONG utest_div64, utest_mod64;
LONGLONG stest_div64, stest_mod64;
UINT utest_div, utest_mod;
INT stest_div, stest_mod;

utest_div64 = udividend64 / divisor;
utest_mod64 = udividend64 % divisor;
DPRINT1("(udiv64) %llu / %lu = %llu, expected 2885681151\n", udividend64, divisor, utest_div64);
DPRINT1("(udiv64) %llu %% %lu = %llu, expected 300\n", udividend64, divisor, utest_mod64);
DbgBreakPoint();

stest_div64 = sdividend64 / sdivisor;
stest_mod64 = sdividend64 % sdivisor;
DPRINT1("(sdiv64) %lld / %lu = %lld, expected -2885681151\n", sdividend64, divisor, stest_div64);
DPRINT1("(sdiv64) %lld %% %lu = %lld, expected -300\n", sdividend64, divisor, stest_mod64);
DbgBreakPoint();

utest_div = udividend / divisor;
utest_mod = udividend % divisor;
DPRINT1("(udiv) %lu / %lu = %u, expected 8\n", udividend, divisor, utest_div);
DPRINT1("(udiv) %lu %% %lu = %u, expected 225\n", udividend, divisor, utest_mod);
DbgBreakPoint();

stest_div = sdividend / sdivisor;
stest_mod = sdividend % sdivisor;
DPRINT1("(sdiv) %ld / %lu = %d, expected -8\n", sdividend, divisor, stest_div);
DPRINT1("(sdiv) %ld %% %lu = %d, expected -225\n", sdividend, divisor, stest_mod);
DbgBreakPoint();
```

And here is WinDbg output:
```
0:000> bp rapps!_rt_udiv64
0:000> g
Breakpoint 0 hit
rapps!_rt_udiv64:
0137100c e92d400f push.w      {r0-r3,lr}
0:000> g
Breakpoint 0 hit
rapps!_rt_udiv64:
0137100c e92d400f push.w      {r0-r3,lr}
0:000> g
(base\applications\rapps\winmain.cpp:80) (udiv64) 1154272460700 / 400 = 0, expected 2885681151
(base\applications\rapps\winmain.cpp:81) (udiv64) 1154272460700 % 400 = 3221225372, expected 300
(fb0.4c): Break instruction exception - code 80000003 (first chance)
ntdll!DbgBreakPoint:
77742c80 defe     __debugbreak
0:000> p
rapps!wWinMain+0x170:
0138614c f7eaff66 bl          rapps!_rt_sdiv64 (0137101c)
0:000> p
rapps!wWinMain+0x174:
01386150 9337     str         r3,[sp,#0xDC]                   008ff774=0056f000
0:000> g
(base\applications\rapps\winmain.cpp:85) (sdiv64) -1154272460700 / 400 = 0, expected -2885681151
(base\applications\rapps\winmain.cpp:86) (sdiv64) -1154272460700 % 400 = -1073741924, expected -300
(fb0.4c): Break instruction exception - code 80000003 (first chance)
ntdll!DbgBreakPoint:
77742c80 defe     __debugbreak
0:000> p
rapps!wWinMain+0x27a:
01386256 f002fbab bl          rapps!__rt_udiv (013889b0)
0:000> p
rapps!wWinMain+0x27e:
0138625a 900d     str         r0,[sp,#0x34]                   008ff6cc=00000000
0:000> g
(base\applications\rapps\winmain.cpp:90) (udiv) 3425 / 400 = 8, expected 8
(base\applications\rapps\winmain.cpp:91) (udiv) 3425 % 400 = 225, expected 225
(fb0.4c): Break instruction exception - code 80000003 (first chance)
ntdll!DbgBreakPoint:
77742c80 defe     __debugbreak
0:000> p
rapps!wWinMain+0x362:
0138633e f002fa97 bl          rapps!__rt_sdiv (01388870)
0:000> p
rapps!wWinMain+0x366:
01386342 900f     str         r0,[sp,#0x3C]                   008ff6d4=00000000
0:000> g
(base\applications\rapps\winmain.cpp:95) (sdiv) -3425 / 400 = -8, expected -8
(base\applications\rapps\winmain.cpp:96) (sdiv) -3425 % 400 = -225, expected -225
(fb0.4c): Break instruction exception - code 80000003 (first chance)
ntdll!DbgBreakPoint:
77742c80 defe     __debugbreak
```

As you can see, udiv/sdiv functions do return proper values, but udiv64/sdiv64 does not.

**Review by learn-more — APPROVED** [review]
Without knowing all to much about the arm architecture, these changes make sense to me.

**Review by binarymaster — COMMENTED** [review]
Confirmed working! Some nits on commits:
- Typo in commit message: `s/__rtsdiv/__rt_sdiv/`
- Please add Jira tickets CORE-17607 and CORE-17614 - they're both will be fixed by this PR
- I'd suggest uppercase `[CRT/ARM]` for commit messages, my personal preference :)

File: sdk/lib/crt/math/arm/__rt_udiv64.s [hunk 1]
```diff
@@ -25,10 +31,8 @@
     mov r0, sp
```
Comment (theR4K) [inline sdk/lib/crt/math/arm/__rt_udiv64.s:31]: I'm also working on fixing that, and you cant transfer all the data thru registers.
In my fixes I used this pattern for the function
__rt_udiv64_worker(
	UINT3264 divisor,
	UINT3264 dividend,
	ARM_DIVRESULT* result)
And send SP thru sp
	mov r4,sp
	push {r4}

File: sdk/lib/crt/math/arm/__rt_sdiv64.s [hunk 2]
```diff
@@ -1,8 +1,9 @@
 /*
- * COPYRIGHT:         BSD - See COPYING.ARM in the top level directory
- * PROJECT:           ReactOS CRT library
- * PURPOSE:           Implementation of __rt_sdiv64
- * PROGRAMMER:        Timo Kreuzer (timo.kreuzer@reactos.org)
+ * PROJECT:     ReactOS CRT library
+ * LICENSE:     MIT (https://spdx.org/licenses/MIT)
+ * PURPOSE:     Implementation of __rt_sdiv64
+ * COPYRIGHT:   Copyright 2015 Timo Kreuzer <timo.kreuzer@reactos.org>
+ *              Copyright 2021 Raman Masanin <rmasanin@ibagroup.eu>
```
Comment (binarymaster) [inline sdk/lib/crt/math/arm/__rt_sdiv64.s:6]: Typo in the name
```suggestion
 *              Copyright 2021 Roman Masanin <rmasanin@ibagroup.eu>
```

Comment (theR4K) [reply sdk/lib/crt/math/arm/__rt_sdiv64.s:6]: I prefer to use 36927roma@gmail.com instead of work email

Comment (tkreuzer) [reply sdk/lib/crt/math/arm/__rt_sdiv64.s]: Fixed

**Review by binarymaster — APPROVED** [review]

**Pull request merged**